ehrapy/api/anndata_ext.py

```diff
@@ -102,6 +102,50 @@ def anndata_to_df(adata: AnnData, add_from_obs: list[str] | str | None = None) -
     return df
 
 
+def move_to_obs(adata: AnnData, to_obs: list[str] | str) -> None:
+    """Move some columns from X to obs inplace
+    Args:
+        adata: The AnnData object
+        to_obs: The columns to move to obs
+
+    Returns:
+        The original AnnData object with moved columns from X to obs
+    """
+    if isinstance(to_obs, str):
+        to_obs = [to_obs]
+    # don't allow moving encoded columns as this could lead to inconsistent data in X and obs
+    if any(column.startswith("ehrapycat") for column in to_obs):
+        raise ObsMoveError("Cannot move encoded columns from X to obs. Either undo encoding or remove them from the list!")
+    indices = adata.var_names.isin(to_obs)
+    df = adata[:, indices].to_df()
+    adata._inplace_subset_var(~indices)
+    adata.obs = adata.obs.join(df)
+    updated_num_uns, updated_non_num_uns = _update_uns(adata, to_obs)
+    adata.uns["numerical_columns"] = updated_num_uns
+    adata.uns["non_numerical_columns"] = updated_non_num_uns
+
+
+def move_to_x(adata: AnnData, to_x: list[str] | str) -> AnnData:
+    """Move some columns from obs to X inplace
+    Args:
+        adata: The AnnData object
+        to_x: The columns to move to X
+
+    Returns:
+        A new AnnData object with moved columns from obs to X. This should not be used for datetime columns currently.
+    """
+    if isinstance(to_x, str):
+        to_x = [to_x]
+    new_adata = concat([adata, AnnData(adata.obs[to_x], dtype="object")], axis=1)
```

Comment:
@Zethson : I noticed the following issue: This will create an `obs` where all columns have `object` dtype regardless whether they are numerical or not. Do you see a way to get this info once its "lost" (best case not by trying to type cast). Or do we need to store this explicitly? Any ideas?

Reply:
I know that this: https://github.com/theislab/anndata/blob/master/anndata/_core/anndata.py#L1188 exists. But this does not really solve the problem here, does it?

Reply:
Yeah not really. We would need like the other way round. I mean would be easy to code ourself, but it would include some sort of type casting, which is what I want to avoid if possible (cause I guess this would be a costly operation).